src/domain/community/virtualContributor/vcProfilePage/VCProfilePage.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import Loading from '../../../../core/ui/loading/Loading';
+import { useUrlParams } from '../../../../core/routing/useUrlParams';
+import { Error404 } from '../../../../core/pages/Errors/Error404';
+import VCPageLayout from '../layout/VCPageLayout';
+import VCProfilePageView from './VCProfilePageView';
+import {
+  useBodyOfKnowledgeProfileQuery,
+  useVirtualContributorQuery,
+} from '../../../../core/apollo/generated/apollo-hooks';
+
+export const VCProfilePage = () => {
+  const { vcNameId = '' } = useUrlParams();
+
+  const { data, loading, error } = useVirtualContributorQuery({
+    variables: {
+      id: vcNameId,
+    },
+  });
+
+  const { data: bokProfile, loading: loadingBok } = useBodyOfKnowledgeProfileQuery({
+    variables: {
+      spaceId: data?.virtualContributor.bodyOfKnowledgeID || '',
```

Comment:
you can use `spaceId: data?.virtualContributor.bodyOfKnowledgeID!` here. normally we just add a comment at the end `spaceId: data?.virtualContributor.bodyOfKnowledgeID! // Ensured by the skip`